syncapi/storage/interface.go

```diff
@@ -104,4 +105,16 @@ type Database interface {
 	StreamEventsToEvents(device *authtypes.Device, in []types.StreamEvent) []gomatrixserverlib.HeaderedEvent
 	// SyncStreamPosition returns the latest position in the sync stream. Returns 0 if there are no events yet.
 	SyncStreamPosition(ctx context.Context) (types.StreamPosition, error)
+	// AddSendToDevice increases the EDU position in the cache and returns the stream position.
+	AddSendToDevice() types.StreamPosition
+	// SendToDeviceUpdatesForSync returns a list of send-to-device updates, after having completed
+	// updates and deletions for previous events. The sync token should be supplied to this function so
+	// that we can clean up old events properly.
+	SendToDeviceUpdatesForSync(ctx context.Context, userID, deviceID string, token types.StreamingToken) ([]types.SendToDeviceEvent, []types.SendToDeviceNID, []types.SendToDeviceNID, error)
```

Comment:
Unclear which token should be supplied, the old token the client gave us or the new calculated token?

Reply:
Also unclear what return order the `[]types.SendToDeviceNID` should be - use named return args here for clarity.